cluster-autoscaler/config/autoscaling_options.go

```diff
@@ -32,6 +32,8 @@ type GpuLimits struct {
 
 // AutoscalingOptions contain various options to customize how autoscaling works
 type AutoscalingOptions struct {
+	// MaxBulkSoftTaint is a number of nodes that can be tainted PreferNoSchedule at the same time.
```

Comment:
The wording does not make it obvious if this is about:
- how many nodes can have a taint at any single time
- or how many nodes can be modified with a taint in one go.

Please change it so it is more understandable.

Reply:
> how many nodes can have a taint at any single time

I don't think the current version limits it at all. Perhaps it should?

Reply:
done

Reply:
@aleksandra-malinowska - we could add it as a second parameter, but how to calculate such limit?

Reply:
It could be a percentage of nodes, and we could keep track of how many we've tainted. But then we'd probably want to taint them in some order (first the empty ones, then candidates), so let's keep it as it is for now and discuss later.

Reply:
This thread reflects all my thoughts:
1. When I saw the field I was 100% sure it was about max number of tainted nodes, not rate limiting. I was very surprised when I read the code.
1. As @aleksandra-malinowska wrote I'm not convinced we should be marking every candidate like that. We know each candidate can be deleted, but that in no way implies all of them can be deleted. In a small cluster majority of nodes can be candidates and tainting most nodes defeats the purpose of tainting them in the first place. Tainting just some candidates requires some persistent ordering of the candidates of course.
@aleksandra-malinowska's ideas sound good to me - percentage of all nodes as a max threshold. Not perfect but consistent order can be achieved by just sorting candidates - already tainted go first, empty second (or should it be the other way round?), followed by the rest. WDYT?

Reply:
>Aleksandra Malinowska's ideas sound good to me - percentage of all nodes as a max threshold. 

Percentage of all nodes seems like a good starting point. Yet I would love have some metric which would tell us in the future if the percentage (or other heuristic) we picked is a good one or should be changed. This could be one of CA KPIs.

Reply:
What about sth like tracking when nodes stop being unneeded according to us? Then we'd know what's the likelihood that given node will be removed in the end, and how taints affect this.